app/angular/src/server/framework-preset-angular-cli.ts

```diff
@@ -1,16 +1,104 @@
-import { Configuration } from 'webpack';
-import { Path } from '@angular-devkit/core';
+import webpack from 'webpack';
 import { logger } from '@storybook/node-logger';
+import TsconfigPathsPlugin from 'tsconfig-paths-webpack-plugin';
 
+import { findAngularProject, readAngularWorkspaceConfig } from './angular-read-workspace';
 import {
-  getAngularCliWebpackConfigOptions,
-  applyAngularCliWebpackConfig,
-} from './angular-cli_config';
+  AngularCliWebpackConfig,
+  extractAngularCliWebpackConfig,
+} from './angular-devkit-build-webpack';
+import { moduleIsAvailable } from './utils/module-is-available';
+import { filterOutStylingRules } from './utils/filter-out-styling-rules';
 
-export function webpackFinal(config: Configuration) {
-  const cwd = process.cwd() as Path;
-  const cliWebpackConfigOptions = getAngularCliWebpackConfigOptions(cwd);
-  logger.info('=> Loading angular-cli config.');
+export async function webpackFinal(baseConfig: webpack.Configuration) {
+  const dirToSearch = process.cwd();
 
-  return applyAngularCliWebpackConfig(config, cliWebpackConfigOptions);
+  if (!moduleIsAvailable('@angular-devkit/build-angular')) {
+    logger.info('=> Using base config because @angular-devkit/build-angular is not installed');
+    return baseConfig;
+  }
+  logger.info('=> Loading angular-cli config');
+
+  // Read angular workspace
+  let workspaceConfig;
+  try {
+    workspaceConfig = await readAngularWorkspaceConfig(dirToSearch);
+  } catch (error) {
+    logger.error(
+      `=> Could not find angular workspace config (angular.json) on this path "${dirToSearch}"`
+    );
+    return baseConfig;
+  }
+
+  // Find angular project
+  let project;
+  try {
+    const fondProject = findAngularProject(workspaceConfig);
+    project = fondProject.project;
+    logger.info(`=> Using angular project '${fondProject.projectName}' for configuring Storybook`);
+  } catch (error) {
+    logger.error(`=> Could not find angular project`);
+    throw error;
```

Comment:
I think I have introduced a change in behavior here 🤔 
I throw a lot more "error" than before 

To avoid the change of behavior it's necessary to ignores the errors if the `project` is not found and if it does not contain `architect.build`. with warn 

but I wonder if this is a good thing 🤷‍♂️

Reply:
If certain setups do not make sense at all I think it's ok to introduce a breaking change. In my experience, misconfiguring storybook may not lead to startup errors but runtime errors instead so it's not even breaking, it is "exit what should not work anyway"